tools/gitleaks/config.toml

```diff
@@ -241,8 +238,53 @@
 	tags = ["key", "API", "generic"]
 	[[rules.Entropies]]
 		Min = "4.5"
-        	Max = "4.7"
-        	Group = "2"
+        Max = "4.7"
+        Group = "2"
+
+[[rules]]
+	description = "Generic Secret"
+	regex = '''(?i)(secret)(.{0,20})?['|"]([0-9a-zA-Z-_\/+!{}/=]{4,120})['|"]'''
+	tags = ["generic", "secret"]
+	[rules.allowlist]
+		regexes = [
+			'''(?i)secret.Data\[\"''',
+			'''(?i)(secret)?(key)?ref''',
+			'''Secret:( *)\"(test|secretName|secret|testns)\"''',
+			'''secret:( *)newKubeConfigSecret\(\"ns1\", \"secret1\"''',
+			'''newSecret\(\"''',
+			'''SecretResourceAttributes''',
+			'''(?i)(pull(_|-)?|CoreV1\(\).|check)Secret(s)?''',
+			'''(?i)Hub(-)?Config(-)?Secret(-)?Name''',
+			'''secret is deleted''',
+			'''secret \(alert/g0\)''',
+			'''secret\', kind: \'value''',
+			'''SecretAccessKey\"] = \"your_aws_secret_access_key\"''',
+			'''Secret\":   \"your_clientSecret\"''',
+			'''secret data under the \"ssh-privatekey\"'''
+		]
+
+[[rules]]
+	description = "Generic User"
+	regex = '''(?i)(user)(.{0,20})?['|"]([0-9a-zA-Z-_\/+!{}/=]{4,120})['|"]'''
+	tags = ["generic", "user"]
+	[rules.allowlist]
+		regexes = [
+			'''userDir = \"/root\"''',
+			'''user_inputs''',
+			'''userNamespace''',
+			'''user_inputs''',
+			'''userNamespace''',
+			'''USER=\$\{ADMIN_USER:-''',
+			'''usermenu''',
+			'''User page\"''',
+			'''user\\\"\", \"state\": ''',
+			'''user\", \"state\": ''',
+			'''User       string `json:\"user\"''',
+			'''username\"] = \"your_username\"''',
+			'''User\", \"Group\"''',
+			'''User\",type=\"string\"''',
+			'''USER \|\| Cypress.env\(\'OPTIONS_HUB_USER'''
```

Comment:
These could be combined into a single regex of the form `(?i)user(foo|bar)`

Trying to think of what is easier from a long term maintenance point of view. Thoughts @pshickeydev 

(applicable to secret rule too)

Reply:
This sounds like a good plan. It's easier to see the repetition when arranged in a column instead of spread across multiple github issues. A proper regular expression could encompass the majority of these with a few special cases (this applies more to the secrets rule).